core/capabilities/ccip/configs/solana/chain_writer.go

```diff
@@ -36,17 +37,16 @@ func getCommitMethodConfig(fromAddress string, routerProgramAddress string, comm
 		},
 		ChainSpecificName: "commit",
 		LookupTables: chainwriter.LookupTables{
-			StaticLookupTables: []solana.PublicKey{
-				commonAddressesLookupTable,
+			DerivedLookupTables: []chainwriter.DerivedLookupTable{
+				getCommonAddressLookupTableConfig(offrampProgramAddress),
 			},
 		},
 		Accounts: []chainwriter.Lookup{
-			getRouterAccountConfig(routerProgramAddress),
+			getOfframpAccountConfig(offrampProgramAddress),
+			getReferenceAddressesConfig(offrampProgramAddress),
 			chainwriter.PDALookups{
-				Name: "SourceChainState",
-				PublicKey: chainwriter.AccountConstant{
-					Address: routerProgramAddress,
-				},
+				Name:      "SourceChainState",
+				PublicKey: offrampAddressConstant(offrampProgramAddress),
 				Seeds: []chainwriter.Seed{
 					{Static: []byte("source_chain_state")},
 					{Dynamic: chainwriter.AccountLookup{Location: merkleRootChainSelector}},
```

Comment:
Could we rename this field to merkleRootSourceChainSelector, to be more clear

Reply:
Seems like we also use this for dest chain state [here](https://github.com/smartcontractkit/chainlink/blob/0630a1b05fc426f56d33724e9b05038d558dc106/core/capabilities/ccip/configs/solana/chain_writer.go#L115) so adding source to the name might be misleading. I didn't touch these extra accounts since these aren't affected by the context changes so I'm trusting these are accurate.

Reply:
So the path "Info.MerkleRoots.ChainSel" always points to the source chain selector.
In Solana, see here: https://github.com/smartcontractkit/chainlink-ccip/blob/fe429267b71335f6bd05371fb5092e3664d31efb/chains/solana/gobindings/ccip_offramp/types.go#L153-L154

In EVM, see here: https://github.com/smartcontractkit/chainlink/blob/6e65deecae053ee1e885da7ce6d1d308364ced1d/core/gethwrappers/ccip/generated/ccip_encoding_utils/ccip_encoding_utils.go#L58-L59

I see you are using it in the "ChainConfigGasPrice" config. I couldn't verify it with onchain code, as the implementation doesn't exist, but please confirm this is what was needed.